invenio_records_lom/utils/util.py

```diff
@@ -0,0 +1,423 @@
+                cursor = cursor[subkey]
+            except (IndexError, TypeError, ValueError) as exc:
+                # `wrapper.get(key, default)` defaults only on errors of type KeyError
+                raise KeyError(dotted_key) from exc
+        return cursor
+
+    def __setitem__(self, dotted_key: str, value: t.Any) -> None:
+        """Set."""
+        if not self.overwritable:
+            if "[]" not in self.split(dotted_key) and dotted_key in self:
+                raise KeyError(
+                    f"Tried to overwrite {dotted_key!r} when overwriting is disabled."
+                )
+
+        cursor = self.data
+        subkeys = self.split(dotted_key)
+        for subkey, next_subkey in zip(subkeys, subkeys[1:] + [None]):
+            # the next key allows to judge what the next value should be
+            next_value = (
+                [] if next_subkey == "[]" else value if next_subkey is None else {}
+            )
+            if subkey == "[]":
+                cursor.append(next_value)
+                cursor = cursor[-1]
+            else:
+                self.check(cursor, subkey)
+                if subkey not in cursor or next_subkey is None:
+                    cursor[subkey] = next_value
+                cursor = cursor[subkey]
+
+    def __delitem__(self, dotted_key: str) -> None:
+        """Delete."""
+        *parent_keys, last_key = self.split(dotted_key)
+        parent = self[".".join(parent_keys)]
+        self.check(parent, last_key)
+        del parent[last_key]  # pylint: disable=unsupported-delete-operation
+
+    def __iter__(self) -> t.Iterator:
+        """Iter."""
+        return iter(self.data)
+
+    def __len__(self) -> None:
+        """Len."""
+        return len(self.data)
+
+    def __repr__(self) -> str:
+        """Repr."""
+        return f"<{type(self).__qualname__}({self.data!r}) at {hex(id(self))}>"
+
+    @staticmethod
+    def check(container: t.Union[dict, list], key: str) -> None:
+        """Check whether `key` fulfills the unambiguity condition."""
+        try:
+            key = int(key)
+        except ValueError:
+            pass
+
+        if isinstance(container, dict) and not isinstance(key, str):
+            raise ValueError("For unambiguity, dict-keys may not be int-castable.")
+
+    @staticmethod
+    def split(dotted_key: str) -> list[t.Union[str, int]]:
+        """Split `dotted_key` into its subkeys."""
+        res = []
+        for subkey in dotted_key.split("."):
+            try:
+                res.append(int(subkey))
+            except ValueError:
+                res.append(subkey)
+        return res
+
+
+def get_learningresourcetypedict() -> dict[str, dict[str, str]]:
+    """Get learningresourcetypes-dict.
+
+    Maps ((url-ending for "https://w3id.org/kim/hcrt/scheme/") -> labels_by_language),
+    where labels_by_language maps (language_code -> label).
+    """
+    path = Path(__file__).resolve().parent / "learning_resource_types.json"
```

Comment:
better: 
```python
parent = Path(__file__).resolve().parent
path = f"{parent}/learning_resource_types.json"
```
the existing versions leads one to believe that division is taking place